packages/browser/__tests__/Session.spec.ts

```diff
@@ -1020,5 +1020,74 @@ describe("Session", () => {
         inIframe: true,
       });
     });
+
+    it("sets the updated session info after silently refreshing", async () => {
+      const sessionId = "mySession";
+      mockLocalStorage({
+        [KEY_CURRENT_SESSION]: sessionId,
+      });
+      mockLocation("https://mock.current/location");
+      const mockedStorage = new StorageUtility(
+        mockStorage({
+          [`${USER_SESSION_PREFIX}:${sessionId}`]: {
+            isLoggedIn: "true",
+          },
+        }),
+        mockStorage({})
+      );
+      const clientAuthentication = mockClientAuthentication({
+        sessionInfoManager: mockSessionInfoManager(mockedStorage),
+      });
+      const validateCurrentSessionPromise = Promise.resolve({
+        issuer: "https://some.issuer",
+        clientAppId: "some client ID",
+        clientAppSecret: "some client secret",
+        redirectUrl: "https://some.redirect/url",
+        tokenType: "DPoP",
+      });
+      clientAuthentication.validateCurrentSession = jest
+        .fn()
+        .mockReturnValue(validateCurrentSessionPromise);
+      const incomingRedirectPromise = Promise.resolve({
+        isLoggedIn: true,
+        webId: "https://some.pod/profile#me",
+        sessionId: "someSessionId",
+      });
+      clientAuthentication.handleIncomingRedirect = jest
+        .fn()
+        .mockReturnValueOnce(incomingRedirectPromise);
+      clientAuthentication.login = jest.fn();
+
+      const windowAddEventListener = jest.spyOn(window, "addEventListener");
+      // ../src/iframe is mocked for other tests,
+      // but we need `setupIframeListener` to actually be executed
+      // so that the callback gets called:
+      const iframeMock = jest.requireMock("../src/iframe");
+      const iframeActual = jest.requireActual("../src/iframe");
+      iframeMock.setupIframeListener.mockImplementationOnce(
+        iframeActual.setupIframeListener
+      );
+      const mySession = new Session({ clientAuthentication }, sessionId);
+
+      // `window.addEventListener` gets called once with ("message", handler)
+      // — get that handler.
+      const messageEventHandler = windowAddEventListener.mock
+        .calls[0][1] as EventListener;
+      const mockedEvent = {
+        origin: window.location.origin,
+        source: null,
+        data: {
+          redirectUrl: "http://arbitrary.com",
+        },
+      } as MessageEvent;
+      // This handler will call `clientAuthentication.handleIncomingRedirect`,
+      // which will return our sessionInfo values:
+      messageEventHandler(mockedEvent);
+
+      await incomingRedirectPromise;
+      await validateCurrentSessionPromise;
+      expect(mySession.info.webId).toBe("https://some.pod/profile#me");
+      expect(mySession.info.sessionId).toBe("someSessionId");
```

Comment:
```suggestion
      expect(mySession.info.sessionId).toBe("someSessionId");
      expect(mySession.info.expirationDate).toBe(961106400);
```
I'm repeating myself but... what a date !

Reply:
4134dc9f